addon/-private/system/model/internal-model.js

```diff
@@ -885,17 +885,28 @@ export default class InternalModel {
     @method clearRelationships
     @private
   */
-  clearRelationships() {
+  clearRelationships(resetAll = false) {
     this.eachRelationship((name, relationship) => {
       if (this._relationships.has(name)) {
         let rel = this._relationships.get(name);
-        rel.clear();
-        rel.removeInverseRelationships();
+
+        if (resetAll === true) {
+          rel.clear();
+          rel.removeInverseRelationships();
```

Comment:
I believe always doing `unloadInverseInternalModel(this)` will nearly fix https://github.com/emberjs/data/pull/5052 the remaining assertion failures will be based on what we decide resurrection + unload + retained by inverse relationship actually means. The code-base is currently slightly confused... specifically:

the following assertion https://github.com/emberjs/data/pull/5052/files#diff-f5099af3b5cad3fc343f3f4b73eb3c97R208  will be true if resurrection does apply in this case (reviving 2 + 6) OR resurrection does not apply and the contacts list will become  ` ['3','4','5','7']`

I think @hjdivad will need to chime in.